### PR TITLE
CID-2065: Update documentation on Limitations on Running Multiple Instances

### PR DESCRIPTION
**Pull request by mohamedlajmileanix**
## 🛠 Changes made
Update documentation on Limitations on Running Multiple Instances of GitLab Broker

## ✨ Type of change
- [x] Documentation update

## 🧪 How Has This Been Tested?

- [x] 👌 No tests required

## 🏎 Checklist:
- [x] My commit message clearly reflects the changes made
- [x] Assigned the appropriate labels (version, PR type, etc.)

### CODE REVIEW

File: docs/github-enterprise-connector.md [hunk 1]
```diff
@@ -96,6 +96,10 @@ docker run --pull=always --restart=always \
 5. Deploy the docker container in your preferred deployment mode (e.g. via K8s, via virtual machine ...)
 6. Shortly after initialization the agent will connect with VSM and you should see logs appearing in the log section of the integration panel in VSM.
 
+#### Limitations on Running Multiple Instances of GitHub Broker
+
+For a given configuration name, we only support a single broker instance to be run at any given time. Any additional instance trying to communicate with VSM will be rejected to prevent race conditions. Please ensure a single instance is running at any time for optimal performance.
```
[the remark this answers is not in the document]
Comment (vg-leanix) [reply docs/github-enterprise-connector.md:101]: @mohamedlajmileanix thanks. my question here: isnt the prevention mechanism here tied to a config set? ie i can have multiple agents per workspace (if and only if they are linked to exactly 1 config set)? put differently i cant have 2 agents for the same config set name.

id vote to be clearer here (hopefully assuming that this is the implemented behaviour as well)

[the remark this answers is not in the document]
Comment (geoandri) [reply docs/github-enterprise-connector.md:101]: Yes @vg-leanix , our backend implementation prevents the broker from receiving an assignment using the `workspaceId`, `configSetName` combination as unique identifier. Given this, I do agree that the documentation should be a bit more clear on this.

[the remark this answers is not in the document]
Comment (vg-leanix) [reply docs/github-enterprise-connector.md:101]: @geoandri - help me understand what this exactly means? do we allow to run multiple instances for any given workspace and config set or not?

[the remark this answers is not in the document]
Comment (geoandri) [reply docs/github-enterprise-connector.md]: Yes, we do allow running multiple instances for any given workspace and config set.

[the remark this answers is not in the document]
Comment (vg-leanix) [reply docs/github-enterprise-connector.md]: @geoandri interesting. does doing so result in the known race conditions?

**Review by geoandri — APPROVED** [review]

**Pull request merged**